internal/fsutil/path.go

```diff
@@ -0,0 +1,47 @@
+package fsutil
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// These functions exists because we work with slash terminated directory paths
+// that come from deb package tarballs but standard library path functions
+// treat slash terminated paths as unclean.
```

Comment:
This text seems like it would belong better in the PR description than in the code. For me at least, I would expect the documentation for the `Dir` function to just contain information about what exactly it does, rather than information about why it is needed. If someone is wondering why the code has used `fsutil.Dir` instead of `filepath.Dir` they could look at the code here and see that it does some extra cleaning.

Reply:
This is subjective. I like the the function comment with examples more. When someone who is familiar with `filepath.Dir()` sees usages of `fsutil.Dir()` the first question they will ask is _Why don't they just use `filepath.Dir`?_ and this comment gives that answer. I don't think the documentation with examples hurts anyone. It's not that verbose to get in the way of reading the code.

Reply:
In my opinion the comments above the functions themselves are enough for this, to me this only serves as an extra block of text to parse and comprehend that doesn't aid my overall understanding